src/malli/core.cljc

```diff
@@ -76,11 +76,31 @@
     (seq children) (into [name] children)
     :else name))
 
+(defn- -transform-if
+  [pred tf]
+  (when tf
+    (fn [x]
+      (if (pred x)
+        (tf x)
+        x))))
+
 (defn- -chain [phase chain]
   (let [f (case phase
             :enter identity
             :leave reverse)]
-    (some->> chain (keep identity) (seq) (f) (reverse) (apply comp))))
+    (some->> chain
+             (keep (fn [item]
+                     (if (vector? item)
+                       (let [pred (first item)
+                             child-chain (second item)]
+                         (assert (fn? pred) "First item in tuple must be a predicate fn")
+                         (assert (seqable? child-chain) "Second item in tuple must be a sequence")
```

Comment:
These assertions are useful because if you call `-chain` with an invalid format, the error becomes very difficult to track down.